# Phase 6: Role specialists for communication

## Who gets shown on the radar chart, and how each name is labeled

Unlike notebooks 0-5, this one doesn't produce evidence for the central
question, it produces a communication artifact: the small set of players
shown on the radar chart (see project brief, section 13), which
illustrates the difference between specialized and multidimensional
profiles. Two categories, never mixed without a label: the statistical
leader of each role within the qualified population (data), and manually
added reference players shown for recognizability even when they don't
lead a role in this specific population (communication choice), per the
decision already logged in `METHODOLOGY.md` under "Role leaders /
reference players."

---

# Fase 6: especialistas por rol para comunicación

## Quién se muestra en el radar chart, y cómo se etiqueta cada nombre

A diferencia de los notebooks 0-5, este no produce evidencia para la
pregunta central, produce un artefacto de comunicación: el pequeño
conjunto de jugadores que se muestra en el radar chart (ver el
planteamiento del proyecto, sección 13), que ilustra la diferencia entre
perfiles especializados y multidimensionales. Dos categorías, que nunca
se mezclan sin etiqueta: el líder estadístico de cada rol dentro de la
población calificada (dato), y jugadores de referencia agregados
manualmente y mostrados por su reconocibilidad aunque no lideren un rol
en esta población específica (decisión de comunicación), según lo ya
decidido en `METHODOLOGY.md` bajo "Líderes por rol / jugadores de
referencia".

In [1]:
import pandas as pd

role_scores = pd.read_csv("../data/processed/role_scores.csv")

role_cols = ["finisher_score", "dribbler_score", "chance_creator_score", "play_organizer_score"]

print("Líder estadístico por rol (población calificada, n=2488):\n")
for role in role_cols:
    leader = role_scores.loc[role_scores[role].idxmax()]
    print(f"{role}: {leader['name']} ({leader[role]:.2f})")
    perfil = leader[role_cols].astype(float).round(2).to_dict()
    print(f"  Perfil completo: {perfil}")
    print()

Líder estadístico por rol (población calificada, n=2488):

finisher_score: Eusébio (100.00)
  Perfil completo: {'finisher_score': 100.0, 'dribbler_score': 97.92, 'chance_creator_score': 84.04, 'play_organizer_score': 79.57}

dribbler_score: Diego Armando Maradona (99.53)
  Perfil completo: {'finisher_score': 88.46, 'dribbler_score': 99.53, 'chance_creator_score': 93.86, 'play_organizer_score': 75.09}

chance_creator_score: Reinhard Libuda (99.75)
  Perfil completo: {'finisher_score': 58.55, 'dribbler_score': 90.08, 'chance_creator_score': 99.75, 'play_organizer_score': 62.99}

play_organizer_score: Isco (99.97)
  Perfil completo: {'finisher_score': 81.94, 'dribbler_score': 98.2, 'chance_creator_score': 77.87, 'play_organizer_score': 99.97}



In [2]:
role_metrics_df = pd.read_csv("../data/processed/role_metrics.csv")

leaders_info = []
for role in role_cols:
    leader = role_scores.loc[role_scores[role].idxmax()]
    minutes = role_metrics_df.loc[role_metrics_df["id"] == leader["id"], "minutesPlayed"].values[0]
    world_cups = role_metrics_df.loc[role_metrics_df["id"] == leader["id"], "world_cups_played"].values[0]
    leaders_info.append({
        "role": role, "name": leader["name"], "score": leader[role],
        "minutesPlayed": minutes, "world_cups_played": world_cups,
    })

leaders_df = pd.DataFrame(leaders_info)
print(leaders_df.to_string(index=False))

                role                   name      score  minutesPlayed  world_cups_played
      finisher_score                Eusébio 100.000000          540.0                  1
      dribbler_score Diego Armando Maradona  99.531083         1940.0                  4
chance_creator_score        Reinhard Libuda  99.746785          363.0                  1
play_organizer_score                   Isco  99.967846          390.0                  1


## A stricter floor for choosing who gets called a "leader"

The organizer-role leader (Isco, 390 minutes) illustrates a real problem:
`totalOppositionHalfPasses` measures passing volume in the opposition
half, not whether that possession created danger, and Spain's 2018
round-of-16 exit to Russia is the textbook case of high-volume, low-threat
possession football. With only 390 minutes, a single match like that
dominates the per-90 rate.

The 270-minute population threshold isn't being changed, it's already
validated with real evidence in Phase 3 and reused throughout. What
changes here is narrower: for choosing which player gets featured as a
role's statistical leader in communication material, a stricter 900-minute
floor is applied (already established in Phase 5 as a meaningful
real-career-sample checkpoint, roughly 10 full matches). Players below
900 minutes stay in the analytical population and every finding already
reported, they're just not eligible to be showcased as "the leader" of a
role for the radar chart.

---

## Un piso más exigente para elegir a quién se llama "líder"

El líder de organizador (Isco, 390 minutos) ilustra un problema real:
`totalOppositionHalfPasses` mide volumen de pase en campo rival, no si esa
posesión generó peligro, y la eliminación de España en octavos del
Mundial 2018 contra Rusia es el caso de libro de fútbol de posesión de
alto volumen y bajo peligro. Con solo 390 minutos, un partido así domina
la tasa per 90.

El umbral poblacional de 270 minutos no se cambia, ya está validado con
evidencia real en la Fase 3 y reutilizado en todo el proyecto. Lo que
cambia acá es más acotado: para elegir qué jugador se muestra como líder
estadístico de un rol en el material de comunicación, se aplica un piso
más exigente de 900 minutos (ya establecido en la Fase 5 como un punto de
corte con sentido real de carrera, aproximadamente 10 partidos completos).
Los jugadores por debajo de 900 minutos siguen en la población analítica
y en todo hallazgo ya reportado, simplemente no son elegibles para
mostrarse como "el líder" de un rol en el radar chart.

In [3]:
print("Columnas disponibles:", role_scores.columns.tolist())

qualified_900 = role_scores[role_scores["minutesPlayed"] >= 900]

print(f"\nJugadores elegibles como líder (900+ minutos): {len(qualified_900)} de {len(role_scores)}\n")

leaders_info = []
for role in role_cols:
    leader = qualified_900.loc[qualified_900[role].idxmax()]
    leaders_info.append({
        "role": role, "name": leader["name"], "score": leader[role],
        "minutesPlayed": leader["minutesPlayed"], "world_cups_played": leader["world_cups_played"],
    })

leaders_df = pd.DataFrame(leaders_info)
print(leaders_df.to_string(index=False))

Columnas disponibles: ['id', 'name', 'finisher_score', 'dribbler_score', 'chance_creator_score', 'play_organizer_score', 'finisher_score_elite_90', 'dribbler_score_elite_90', 'chance_creator_score_elite_90', 'play_organizer_score_elite_90', 'n_elite_roles_90', 'finisher_score_elite_95', 'dribbler_score_elite_95', 'chance_creator_score_elite_95', 'play_organizer_score_elite_95', 'n_elite_roles_95', 'finisher_score_elite_99', 'dribbler_score_elite_99', 'chance_creator_score_elite_99', 'play_organizer_score_elite_99', 'n_elite_roles_99', 'finisher_score_elite_85', 'dribbler_score_elite_85', 'chance_creator_score_elite_85', 'play_organizer_score_elite_85', 'n_elite_roles_85', 'role_score_range', 'role_score_std', 'role_score_avg', 'minutesPlayed', 'world_cups_played']

Jugadores elegibles como líder (900+ minutos): 258 de 2488

                role                   name     score  minutesPlayed  world_cups_played
      finisher_score            David Villa 98.794212          901.0        

In [4]:
reference_pairs = {
    "play_organizer": ["Xavi", "Andrea Pirlo"],
    "chance_creator": ["Zinedine Zidane", "Johan Cruyff"],
    "dribbler": ["Neymar", "Jairzinho"],
    "finisher": ["Cristiano Ronaldo", "Kylian Mbappé"],
}

all_references = [name for pair in reference_pairs.values() for name in pair]

reference_df = role_scores[role_scores["name"].isin(all_references)][
    ["name", "minutesPlayed", "world_cups_played"] + role_cols
]
print(reference_df.to_string(index=False))

found = reference_df["name"].tolist()
missing = [n for n in all_references if n not in found]
print(f"\nNo encontrados en la población (revisar nombre exacto o si no llega a 270 min): {missing}")

             name  minutesPlayed  world_cups_played  finisher_score  dribbler_score  chance_creator_score  play_organizer_score
Cristiano Ronaldo         2206.0                  6       96.637192       83.252947             61.057074             37.138264
     Andrea Pirlo          972.0                  3       62.379421       62.855038             85.361736             97.652733
             Xavi         1124.0                  4       30.727492       65.702036             91.318328             99.485531
  Zinedine Zidane         1109.0                  3       86.870311       86.334405             88.975080             92.564309
           Neymar         1225.0                  4       98.124330       96.128081             92.954180             78.754019
    Kylian Mbappé         1829.0                  3       98.298499       84.539121             78.400322             44.935691
        Jairzinho         1430.0                  3       85.008039       98.995177             72.62861

In [5]:
statistical_leaders = leaders_df.copy()
statistical_leaders["role_context"] = statistical_leaders["role"].str.replace("_score", "", regex=False)
statistical_leaders = statistical_leaders.drop(columns=["role"])
statistical_leaders["label"] = "statistical_leader"

reference_rows = []
for role, names in reference_pairs.items():
    for name in names:
        row = role_scores[role_scores["name"] == name].iloc[0]
        reference_rows.append({
            "role_context": role,
            "name": name,
            "score": row[f"{role}_score"],
            "minutesPlayed": row["minutesPlayed"],
            "world_cups_played": row["world_cups_played"],
        })

reference_leaders = pd.DataFrame(reference_rows)
reference_leaders["label"] = "reference_addition"

role_specialists = pd.concat([statistical_leaders, reference_leaders], ignore_index=True)
print(role_specialists.sort_values(["role_context", "label"], ascending=[True, False]).to_string(index=False))

                  name     score  minutesPlayed  world_cups_played   role_context              label
         David Beckham 98.295820         1028.0                  3 chance_creator statistical_leader
       Zinedine Zidane 88.975080         1109.0                  3 chance_creator reference_addition
          Johan Cruyff 98.689711          630.0                  1 chance_creator reference_addition
Diego Armando Maradona 99.531083         1940.0                  4       dribbler statistical_leader
                Neymar 96.128081         1225.0                  4       dribbler reference_addition
             Jairzinho 98.995177         1430.0                  3       dribbler reference_addition
           David Villa 98.794212          901.0                  3       finisher statistical_leader
     Cristiano Ronaldo 96.637192         2206.0                  6       finisher reference_addition
         Kylian Mbappé 98.298499         1829.0                  3       finisher reference

In [6]:
role_specialists.to_csv("../data/processed/role_specialists.csv", index=False)
print(f"Guardado: {role_specialists.shape[0]} filas, {role_specialists.shape[1]} columnas")

Guardado: 12 filas, 6 columnas


## Top 10 by role, for reference

Not used elsewhere in the pipeline, kept here simply as a visual record
of what the top of each role distribution looks like, at both the
270-minute analytical threshold and the stricter 900-minute floor used to
pick this notebook's statistical leaders.

---

## Top 10 por rol, de referencia

No se usa en el resto del pipeline, se deja acá simplemente como registro
visual de cómo se ve el top de la distribución de cada rol, tanto en el
umbral analítico de 270 minutos como en el piso más exigente de 900
minutos usado para elegir a los líderes estadísticos de este notebook.

In [7]:
qualified_900 = role_scores[role_scores["minutesPlayed"] >= 900]

for role in role_cols:
    print(f"\n=== Top 10: {role} (270+ min) ===")
    top10_270 = role_scores.nlargest(10, role)[["name", role, "minutesPlayed", "world_cups_played"]]
    print(top10_270.to_string(index=False))

    print(f"\n=== Top 10: {role} (900+ min) ===")
    top10_900 = qualified_900.nlargest(10, role)[["name", role, "minutesPlayed", "world_cups_played"]]
    print(top10_900.to_string(index=False))


=== Top 10: finisher_score (270+ min) ===
                name  finisher_score  minutesPlayed  world_cups_played
             Eusébio      100.000000          540.0                  1
Preben Elkjær Larsen       99.691854          315.0                  1
    Valeriy Porkuyan       99.323419          270.0                  1
     Christian Vieri       98.954984          818.0                  2
         David Villa       98.794212          901.0                  3
   Gabriel Batistuta       98.767417          988.0                  3
         Gerd Müller       98.673633         1230.0                  2
       Karim Benzema       98.640139          450.0                  1
     Oliver Bierhoff       98.579850          526.0                  2
         Ferenc Bene       98.512862          360.0                  1

=== Top 10: finisher_score (900+ min) ===
             name  finisher_score  minutesPlayed  world_cups_played
      David Villa       98.794212          901.0                 

## Phase 6 conclusion

Built the reference set for the radar chart: one statistical leader per
role (David Villa, Maradona, Beckham, Rodri, all with a 900-minute floor
applied specifically for this selection, stricter than the 270-minute
analytical population, after Isco's 390-minute organizer lead turned out
to be driven by Spain's low-threat, high-volume passing display in the
2018 round of 16 against Russia) plus two manually added reference
players per role (recognizable names shown for context, not for leading
that metric in this population). Every reference player was checked
against their own four scores before being assigned to a role: Ronaldinho
was initially considered for dribbler but dropped once his own data
showed chance creator, not dribbling, as his highest score, replaced with
Jairzinho, whose dribbler score is unambiguously his own career peak.

**Result:** `data/processed/role_specialists.csv`, 12 rows, explicitly
labeled `statistical_leader` or `reference_addition`, ready to feed the
Tableau radar chart without ambiguity between data and communication
choices.

---

## Conclusión de la Fase 6

Se construyó el conjunto de referencia para el radar chart: un líder
estadístico por rol (David Villa, Maradona, Beckham, Rodri, todos con un
piso de 900 minutos aplicado específicamente para esta selección, más
exigente que la población analítica de 270 minutos, después de que el
liderazgo de Isco en organizador con 390 minutos resultara estar impulsado
por la actuación de España de mucho volumen y poco peligro en los octavos
de 2018 contra Rusia) más dos jugadores de referencia agregados
manualmente por rol (nombres reconocibles mostrados por contexto, no por
liderar esa métrica en esta población). Cada jugador de referencia se
revisó contra sus propios cuatro puntajes antes de asignarlo a un rol:
Ronaldinho se consideró inicialmente para desequilibrante pero se
descartó cuando sus propios datos mostraron que su puntaje más alto era
creador de juego, no regate, reemplazado por Jairzinho, cuyo puntaje de
desequilibrante es inequívocamente su propio pico de carrera.

**Resultado:** `data/processed/role_specialists.csv`, 12 filas,
etiquetadas explícitamente como `statistical_leader` o
`reference_addition`, listas para alimentar el radar chart de Tableau sin
ambigüedad entre dato y decisión de comunicación.